src/client-handl/index.ts

```diff
@@ -0,0 +1,718 @@
+   * entity cache.
+   *
+   */
+  public async getDataEntityCacheSize(): Promise<number> {
+    try {
+      return this._cache.dataEntities.size();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Given a valid key, the method returns the matching entry
+   * from the query path cache.
+   *
+   */
+  public async getQueryPathCacheEntry(key: string): Promise<any> {
+    try {
+      return this._cache.queryPaths.get(key);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * The method returns the number of entries in the query
+   * path cache.
+   *
+   */
+  public async getQueryPathCacheSize(): Promise<number> {
+    try {
+      return this._cache.queryPaths.size();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Given a valid key, the method returns the matching entry
+   * from the response cache.
+   *
+   */
+  public async getResponseCacheEntry(key: string): Promise<ResponseCacheEntryResult | undefined> {
+    try {
+      const entry = await this._cache.responses.get(key);
+      if (!entry) return undefined;
+      const { cacheMetadata, data } = entry;
+      return { cacheMetadata: createCacheMetadata({ cacheMetadata }), data };
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * The method returns the number of entries in the response cache.
+   *
+   */
+  public async getResponseCacheSize(): Promise<number> {
+    try {
+      return this._cache.responses.size();
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * The method imports data exported by another handl
+   * into the caches.
+   *
+   */
+  public async importCaches(caches: ExportCachesResult): Promise<void> {
+    try {
+      return this._cache.import(caches);
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * The method makes query, mutation and subscription requests and
+   * handles request parsing, filtering and caching.
+   *
+   */
+  @clockAsyncMethod
+  public async request(query: string, opts: RequestOptions = {}): Promise<RequestResult> {
+    let errors: Error[] = [];
+
+    if (!isString(query)) {
+      errors.push(new TypeError("Request expected query to be a string."));
+    }
+
+    if (!isPlainObject(opts)) {
+      errors.push(new TypeError("Request expected opts to be a plain object."));
+    }
+
+    if (errors.length) return Promise.reject(errors);
+
+    if (opts.fragments && (!isArray(opts.fragments) || opts.fragments.some((value) => !isString(value)))) {
+      return Promise.reject(new TypeError("Request expected opts.fragments to be an array of strings."));
+    }
+
+    try {
+      const context: RequestContext = { fieldTypeMap: new Map() };
+      const updated = await this._requestParser.updateRequest(query, opts, context);
+      errors = validate(this._schema, updated.ast) as GraphQLError[];

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)